File: dotnet/azure/Mcma.Azure.BlobStorage/Proxies/BlobStorageProxy.cs
using Microsoft.Azure.Storage.Blob;

namespace Mcma.Azure.BlobStorage.Proxies
{
    public abstract class BlobStorageProxy<T> where T : BlobStorageLocator
    {
        protected BlobStorageProxy(T locator, CloudBlobClient client)
        {
            Locator = locator;
            Client = client;
            Container = Client.GetContainerReference(Locator.Container);
        }

        public T Locator { get; }

        public CloudBlobClient Client { get; }

        public CloudBlobContainer Container { get; }
    }
}

File: nodejs/azure/blob-storage/lib/proxies/blob-storage-proxy.ts
import { BlobServiceClient, ContainerClient, StorageSharedKeyCredential, HttpRequestBody } from "@azure/storage-blob";
import { BlobStorageLocator } from "../blob-storage-locator";

export abstract class BlobStorageProxy<T extends BlobStorageLocator> {
    public container: ContainerClient;
    protected credential: StorageSharedKeyCredential;

    constructor(public locator: T, public client: BlobServiceClient) {
        this.container = this.client.getContainerClient(locator.container);

        this.credential = this.client.credential as StorageSharedKeyCredential;
    }
}

